pkg/pillar/types/watcher.go

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) 2017 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package types
+
+import (
+	"github.com/lf-edge/eve/pkg/pillar/base"
+	uuid "github.com/satori/go.uuid"
+)
+
+// UsageZone :
+type UsageZone int
+
+const (
+	// GreenZone :
+	GreenZone UsageZone = 0
+	// YellowZone :
+	YellowZone UsageZone = 1
+	// OrangeZone :
+	OrangeZone UsageZone = 2
+	// RedZone :
+	RedZone UsageZone = 3
+)
+
+// MemoryNotification :
+type MemoryNotification struct {
+	Total uint64 // Total memory in Bytes
+	Used  uint64 // Used memory in Bytes
+	Zone  UsageZone
+
+	PrevUsage uint64   // Previous (last) memory usage in Bytes
+	LastFive  []uint64 // Last 5 Usage percentage values
```

Comment:
Would be good to comment that lastFive is in the reverse order.

Reply:
Done